src/main/java/Storage.java

```diff
@@ -19,18 +18,19 @@ public static void writeToFile(ArrayList<Task> tasks) throws IOException {
         System.out.println("Auto-saved.");
         fw.close();
     }
-    public static ArrayList<Task> loadFile() throws IOException {
-        file = new File("src/main/java/duke.txt");
+    public TaskList loadFile() throws IOException {
+        //file = new File("src/main/java/duke.txt");
 
         if (!file.exists()) {
             file.createNewFile();
         }
 
         BufferedReader reader = new BufferedReader(new FileReader("src/main/java/duke.txt"));
         String str;
-        ArrayList<Task> loadedList = new ArrayList<>();
+        TaskList loadedList = new TaskList();
 
         while((str = reader.readLine()) != null) {
+
             if(str.charAt(1) == 'T') {
```

Comment:
Perhaps try to be consistent when using enums? Maybe save 'T' / 'X', 'D', etc. as some sort of character enum